frontend/src/store/modules/quickLink.ts

```diff
@@ -0,0 +1,110 @@
+import { flatten } from "lodash-es";
+import {
+  GalleryHorizontalEndIcon,
+  SettingsIcon,
+  CircleDotIcon,
+} from "lucide-vue-next";
+import { computed, h, type VNode } from "vue";
+import { t } from "@/plugins/i18n";
+import {
+  WORKSPACE_ROUTE_LANDING,
+  WORKSPACE_ROUTE_USERS,
+  WORKSPACE_ROUTE_SQL_REVIEW,
+} from "@/router/dashboard/workspaceRoutes";
+import { useCurrentUserV1 } from "@/store";
+import {
+  useDynamicLocalStorage,
+  useDashboardSidebar,
+  type DashboardSidebarItem,
+} from "@/utils";
+
+export interface QuickLink {
+  id: string;
+  title: string;
+  route?: string;
+  icon: () => VNode;
+}
+
+export function useQuickLink() {
+  const currentUser = useCurrentUserV1();
+  const { dashboardSidebarItemList } = useDashboardSidebar();
+
+  const getAccessListBySidebar = (
+    sidebarItem: DashboardSidebarItem,
+    icon: () => VNode = () => h(SettingsIcon)
+  ): QuickLink[] => {
+    if (sidebarItem.type === "route" && sidebarItem.title && sidebarItem.name) {
+      if (sidebarItem.name === WORKSPACE_ROUTE_LANDING) {
+        return [];
+      }
+      return [
+        {
+          id: sidebarItem.name,
+          title: sidebarItem.title,
+          route: sidebarItem.name,
+          icon: sidebarItem.icon || icon,
+        },
+      ];
+    }
+
+    if (sidebarItem.children) {
+      return flatten(
+        sidebarItem.children.map((child) =>
+          getAccessListBySidebar(child, sidebarItem.icon)
+        )
+      ).filter((item) => item) as QuickLink[];
+    }
+
+    return [];
+  };
+
+  const fullQuickLinkList = computed((): QuickLink[] => {
+    const accessList = flatten(
+      dashboardSidebarItemList.value.map((item) => getAccessListBySidebar(item))
+    ).filter((item) => item) as QuickLink[];
+
+    accessList.unshift(
+      {
+        id: "visit-projects",
+        title: t("landing.quick-link.visit-prjects"),
+        icon: () => h(GalleryHorizontalEndIcon),
+      },
+      {
+        id: "visit-issues",
+        title: t("landing.quick-link.visit-issues"),
+        icon: () => h(CircleDotIcon),
+      }
+    );
+
+    return accessList;
+  });
+
+  const quickAccessConfig = useDynamicLocalStorage<string[]>(
+    computed(() => `bb.quick-access.${currentUser.value.name}`),
+    [
+      "visit-issues",
```

Comment:
Per offline discussion with @d-bytebase, removing "My Issues" from the dashboard sidebar might cause confusion for old users.
However, we also want a simple "Issues" page like GitHub, for example:

![CleanShot 2024-11-26 at 17 43 53@2x](https://github.com/user-attachments/assets/934cfb1f-68c9-430d-8645-0fae613bd159)

It has a different layout from the previous "My Issues" page. So I'm thinking maybe we can remove "My Issues" from the sidebar, but add the quick link on the landing page:

![CleanShot 2024-11-26 at 17 46 37@2x](https://github.com/user-attachments/assets/81878283-6958-4a34-8035-a24bda160f36)

Any suggestions?

Reply:
+1

Reply:
How about adding a label `My Issues` to the issues button, like the SQL editor button? So that the issues will be more noticeable after changes.

<img width="180" alt="image" src="https://github.com/user-attachments/assets/34a1c05b-546b-4621-8746-55b1aa2f5dcb">

Reply:
LGTM

Reply:
![CleanShot 2024-11-27 at 10 49 16@2x](https://github.com/user-attachments/assets/ea05a3fb-b539-4576-8335-3475c21806ba)